### PR TITLE
dropdown fix & fixed input and select generic styles

### PR DESCRIPTION
**Pull request by Divinare**
Resolves: #241 

About the PR:
* Dropdown uses now select & option HTML elements
* Added generic input and select styles

### CODE REVIEW

File: src/components/controls/Dropdown.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import * as React from 'react';
 import * as s from './dropdown.scss';
-import DownArrow from '../../icons/downArrow';
 
 interface IDropdownState {
-    isOpen: boolean;
+    selectedName?: string;
+    selectedValue?: string;
```
Comment (neodyymi) [inline src/components/controls/Dropdown.tsx]: Why are these separate? It seems like selectedName is never used.

File: src/components/controls/Dropdown.tsx [hunk 2]
```diff
@@ -17,67 +17,42 @@ class Dropdown extends React.Component
     constructor(props: any) {
         super(props);
         this.state = {
-            isOpen: false,
+            selectedName: undefined,
+            selectedValue: undefined,
```
Comment (neodyymi) [inline src/components/controls/Dropdown.tsx]: Why are they undefined when selected is a mandatory prop.

File: src/components/controls/dropdown.scss [hunk 3]
```diff
@@ -1,53 +1,24 @@
...
-    }
+.dropdownView {
+    height: 43px; /* TODO: fix this hack to get dropdown height the same as input field height */
+    margin-top: 5px;
+
+    background:
+        linear-gradient(45deg, transparent 50%, $busBlue 50%),
+        linear-gradient(135deg, $busBlue 50%, transparent 50%);
+    background-position:
+        calc(100% - 21px) calc(16px + 2px),
```
Comment (neodyymi) [inline src/components/controls/dropdown.scss]: 16+2 = 18

File: src/components/sidebar/linkView/LinkView.tsx [hunk 4]
```diff
@@ -121,13 +121,11 @@ class LinkView extends React.Component<ILinkViewProps, ILinkViewState> {
                         />
                     </div>
                     <div className={s.inputContainer}>
```
Comment (neodyymi) [inline src/components/sidebar/linkView/LinkView.tsx]: `<InputContainer>` and className={s.inputContainer} seems confusing..

Comment (neodyymi) [reply src/components/sidebar/linkView/LinkView.tsx]: Could we just have these dropdowns include the inputcontainer part inside them? With an optional label for the cases where we need "KUTSU-/JÄTTÖ-...."

File: src/components/sidebar/linkView/LinkView.tsx [hunk 5]
```diff
@@ -170,37 +166,31 @@ class LinkView extends React.Component<ILinkViewProps, ILinkViewState> {
...
-                        <Dropdown
-                            onChange={this.onChange}
-                            items={['Kyllä']}
-                            selected={'Ei'}
-                        />
-                    </div>
+                    <Dropdown
+                        onChange={this.onChange}
+                        items={['Kyllä']}
+                        selected={'Ei'}
```
Comment (neodyymi) [inline src/components/sidebar/linkView/LinkView.tsx]: How can 'Ei' be selected if items doesn't contain it?

File: src/components/sidebar/linkView/LinkView.tsx [hunk 6]
```diff
@@ -121,13 +121,11 @@ class LinkView extends React.Component<ILinkViewProps, ILinkViewState> {
                         />
```
Comment (neodyymi) [inline src/components/sidebar/linkView/LinkView.tsx]: Why sometimes use `<InputContainer>` and other times `<div>` and `<input>`?

File: src/components/controls/Dropdown.tsx (hunk 2 above)
Comment (Divinare) [reply src/components/controls/Dropdown.tsx]: If it is not undefined, if I set it as '', there is a warning at console:
Warning: Select elements must be either controlled or uncontrolled (specify either the value prop, or the defaultValue prop, but not both). Decide between using a controlled or uncontrolled select element and remove one of these props

File: src/components/sidebar/linkView/LinkView.tsx (hunk 4 above)
Comment (Divinare) [reply src/components/sidebar/linkView/LinkView.tsx]: We could have label for dropdown container but then we should use labels for dropdowns as well

Comment (Divinare) [reply src/components/sidebar/linkView/LinkView.tsx]: Made a new card to implement this fix: https://github.com/HSLdevcom/jore-map-ui/issues/283

File: src/components/sidebar/linkView/LinkView.tsx (hunk 6 above)
Comment (Divinare) [reply src/components/sidebar/linkView/LinkView.tsx]: Not part of this PR. This is only for fixing dropdown. See card #283

File: src/components/sidebar/linkView/LinkView.tsx (hunk 5 above)
Comment (Divinare) [reply src/components/sidebar/linkView/LinkView.tsx]: This isn't the final solution yet, only a static page. And this is how it was before

**Comment by Divinare** [comment]
Review fixes done

File: src/components/controls/Dropdown.tsx (hunk 2 above)
Comment (neodyymi) [reply src/components/controls/Dropdown.tsx]: Could it not be set as `props.selected` ?

File: src/components/controls/Dropdown.tsx (hunk 1 above)
Comment (neodyymi) [reply src/components/controls/Dropdown.tsx]: fixed in commit b1d78c7

File: src/components/sidebar/linkView/LinkView.tsx (hunk 4 above)
Comment (neodyymi) [reply src/components/sidebar/linkView/LinkView.tsx]: Sounds good

File: src/components/sidebar/linkView/LinkView.tsx (hunk 6 above)
Comment (neodyymi) [reply src/components/sidebar/linkView/LinkView.tsx]: 👍

**Pull request merged**